myJoin.pl

```diff
@@ -52,22 +69,30 @@
 #Duplicate line names removed, keys can not be duplicated in hash.
 #increment any names repeated more than once, indicates they are duplicates.
 
+#subroutine to find duplicates within and between files
 sub line_hash1{
+    my %observed;
     my $input = shift;
     open(FILE, $input) or die $!;
     if ($header){
         $header_text = <FILE>;
         chomp $header_text;
-    }
-    while (<FILE>){
-        chomp;
-        $line1{$_} += 1;
-    }
+        }
+        while (<FILE>){
+            chomp;
+            #hash to remove duplicate names from individual file
+            $observed{$_}++;
+            next if $observed{$_} > 1;
```

Comment:
I think that you want `next if $observed{$_} >= 1;` Otherwise, the first TWO observations in a file will continue past this line.